common/djangoapps/course_modes/views.py

```diff
@@ -260,7 +260,7 @@ def get(self, request, course_id, error=None):  # lint-amnesty, pylint: disable=
         # and happy path version is ready to be rolled out to all users.
         if VALUE_PROP_TRACK_SELECTION_FLAG.is_enabled():
             if not error:  # TODO: Remove by executing REV-2355
-                if not enterprise_customer:  # TODO: Remove by executing REV-2342
+                if not enterprise_customer_for_request(request):  # TODO: Remove by executing REV-2342
```

Comment:
Would be valuable to add some comments into why we encountered the error for the specific courses once you do figure out the reason